### ISSUE TITLE
Resource Limitations to Transforms

### ISSUE THREAD

**Issue by Sarcose** [ToCraft/woodwalkers-mod#159]
### Describe your idea, include images if necessary

If this would be better as an addon I would love your suggestion how it could be implemented. I'd like to set up some form of limitation to the transformation mechanic to force players to think strategically when they have large lists of powerful mobs to choose from (our server is using Remorphed, and has already enabled `revoke2ndshapeondeath`):
- Configurably hunger gets expended while transformed. Moreso while flying. Possibly Mob health dependent. Possibly able to be mitigated by something.
- Or, a resource must be consumed (emerald held in hand for instance) in order to transform

The reason the suggestion is so vague is because I am not sure the easiest or best way to limit the mechanic so I was hoping for at least one option that does this. Would a datapack that assigns a hunger-consuming trait to every mob be the simplest way to do this?

Our server is Minecraft 1.21, Fabric 1.16.0

Thank you

**Comment by ToCraft** [comment]
Yep, it would break the intended gameplay, therefore an addon would be better ;D

You could implement the hunger using [Tick Handler](https://github.com/ToCraft/woodwalkers-mod/wiki/Start-Coding#tick-handler) and the resource limit by using [Shape Events](https://github.com/ToCraft/woodwalkers-mod/wiki/Start-Coding#shape-events) and where the player inventory is checked and/or modified.
